Clarify use of getWorkKeys in documentation

maccabeelevine opened the pull request:
MARC 240 has been listed as a non-uniform title field since #1581 introduced record versions, and I can't find any discussion about it specifically in that PR, but [LOC](https://www.loc.gov/marc/bibliographic/bd240.html) (and others) say 240 is a uniform title field.

maccabeelevine committed:
Treat MARC 240 as a uniform title field

maccabeelevine commented:
I noticed this after we made a local override of WorkKeys.java to disable the uniform title part altogether, as we do not want translated titles to be linked together (for now), and they were still being linked via the 240.  My local cataloger colleagues  @lmccoll44 and @gae214 could not find a reason 240 would be configured as non-uniform.

Assuming it was not accidental, my only guess is that the reason was to achieve some effect of the non-uniform [title + author logic](https://github.com/vufind-org/vufind/blob/dev/import/index_java/src/org/vufind/index/WorkKeys.java#L81).  But not clear what.

Of course it is simple to fix this locally in marc_local.properties, but I figured it was worth fixing for others.  That said, it's an obvious breaking change.

demiankatz commented:
I don't have strong opinions about this, so I'll wait and see what @EreMaijala has to say. :-)

EreMaijala commented:
240 is a uniform title field, but it's used when there's also a 100, 110 or 111 field. This means that we can create an author+title (AT) key from it, which makes it more unique than just a uniform title (UT) key. So, while 240 is indeed a uniform title, it can be used like a title here. I wouldn't change the default here, but I'm still open to argument. :) Or adding a clarifying comment.

EreMaijala reviewed (changes requested):
240 is a uniform title field, but it's used when there's also a 100, 110 or 111 field. This means that we can create an author+title (AT) key from it, which makes it more unique than just a uniform title (UT) key. So, while 240 is indeed a uniform title, it can be used like a title here. I wouldn't change the default here, but I'm still open to argument. :) Or adding a clarifying comment.

maccabeelevine committed:
Add clarifying comment

maccabeelevine commented:
@EreMaijala Ok that's helpful.  If I understand then, the distinction here

> Parameters:
> uniform title fields (field spec)
> title fields (field spec)
> title fields with non-filing characters removed (field spec)
> ...

is not really between uniform title vs title fields, but between fields which should become work keys on their own vs. those that should be combined with authors (in every possible combination) to become work keys.  And this distinction is similar, but not the same, as which ones are uniform title fields.

This all hurts my brain.  I tried providing a "clarifying comment" but it's possible that it makes it even more confusing.  And if I'm the only one who has noticed the oddity about 240 so far, maybe it is better to leave the wording (and defaults) as they are and just close the PR.

demiankatz commented:
@maccabeelevine, @EreMaijala is out of office until August, so this conversation will have to wait a little bit. I think it's probably worth adding a comment whether or not we change anything, but there's also the question of where to put that comment -- marc.properties, marc_local.properties, both places, or in detail in one place with a reference in the other?

EreMaijala commented:
@maccabeelevine If you stumbled on it, someone else could too. So a clarifying comment would make sense. I'd put it at least into marc_local.properties since that's what you'd edit locally, but then maybe it also needs to be in marc.properties so that anyone considering changes to the defaults has a chance to see it too.

demiankatz commented:
Thanks, @EreMaijala. How about we add a comment to marc.properties simply saying `See marc_local.properties for an explanation of the getWorkKeys parameters` so we don't have to maintain the documentation in two places. Then we can simply adjust the comments in marc_local.properties to be a little more clear. (We might also want to adjust WorkKeys.java to match, depending on the nature of our changes).

That just leaves the final question: exactly how should we change the comments? My thought is that maybe saying that the first parameter is for uniform titles is misleading. Maybe better to say something like `Field specification to use for generating uniform title keys (only includes title-only uniform title fields; e.g. 240 is better suited for generating author/title keys and is treated as a regular title key."

What does everyone think?

In any case, I'm setting a milestone of 10.1 and making this a documentation PR so we don't lose track of the work (and can discuss in a future Community Call if that would be helpful).

EreMaijala commented:
@demiankatz I like your proposal for the comment. It's short, clear and conveys all the important bits of information.

maccabeelevine committed:
Restore 240 to non-uniform title

maccabeelevine committed:
Document actual behavior of work keys specs

maccabeelevine committed:
Tweak comment

maccabeelevine committed:
Make comment grammar consistent

maccabeelevine committed:
Remove whitespace

maccabeelevine committed:
Add back whitespace

maccabeelevine committed:
No, really

maccabeelevine commented:
Ok, I finally got back to this.  I reverted the field sped changes and added a clarifying comment to marc_local.properties (and a reference from marc.properties).  I tweaked @demiankatz 's suggested language a bit to make more sense to my own brain, framing it as how each field spec will be used as opposed to what it is theoretically for.  I hope it still makes sense.

maccabeelevine committed:
Shorten wording

demiankatz commented on import/marc_local.properties:
```diff
@@ -78,8 +78,12 @@
 # Parameters:
 #   uniform title fields (field spec)
+#       Note: This field specification is for generating uniform title keys;
+#       its fields will be used solo.
```
I'm not sure if "used solo" is 100% clear -- would it be better to add a few more words, like:

"its fields will be used solo, not combined to make author/title keys"

? Or is that not what you mean?

maccabeelevine replied:
Yes that's a good improvement -- done!

demiankatz reviewed (changes requested):
I think we're almost there, @maccabeelevine -- just one line that I wasn't sure I fully understood.

maccabeelevine committed:
Clarify 'solo'

demiankatz reviewed (approved):
Thanks, @maccabeelevine!
